library/core/src/slice/iter.rs

```diff
@@ -2348,7 +2384,12 @@ impl<'a, T> Iterator for RChunks<'a, T> {
             None
         } else {
             let chunksz = cmp::min(self.v.len(), self.chunk_size);
-            let (fst, snd) = self.v.split_at(self.v.len() - chunksz);
+            // SAFETY: split_at_unchecked just requires the argument be less
+            // than the length. This could only happen if the expression
+            // `self.v.len() - chunksz` overflows. This could only happen if
+            // `chunksz > self.v.len()`, which is impossible as we initialize it
+            // as the `min` of `self.v.len()` and `self.chunk_size`.
+            let (fst, snd) = unsafe { self.v.split_at_unchecked(self.v.len() - chunksz) };
```

Comment:
Maybe the `self.v.len()` could be lifted and put in front of the `cmp::min`, then it's a single equation and we can talk directly about its results.

Reply:
I did this but I'm not really sure it helps.